src/lib/core/fiber.c

```diff
@@ -1607,6 +1618,8 @@ struct cord_thread_arg
 void *cord_thread_func(void *p)
 {
 	struct cord_thread_arg *ct_arg = (struct cord_thread_arg *) p;
+	if (is_shutting_down)
```

Comment:
That looks very strange. What is the flag gets set right after you check it here? I don't see any synchronization here with the main thread. Moreover you access a plain boolean in more than one thread without any protection like atomic-exchange or atomic-load, that doesn't look right.

If a thread is started after shutdown, I suppose it should be properly joined anyway. Maybe without event loop, if it is already stopped and our join depends on it. Then you still can do `pthread_join()` or whatever we use underneath.